lib/chef/provider/package/windows.rb

```diff
@@ -36,19 +37,23 @@ class Windows < Chef::Provider::Package
 
         # load_current_resource is run in Chef::Provider#run_action when not in whyrun_mode?
         def load_current_resource
-          @new_resource.source(Chef::Util::PathHelper.validate_path(@new_resource.source))
-
           @current_resource = Chef::Resource::WindowsPackage.new(@new_resource.name)
-          @current_resource.version(package_provider.installed_version)
-          @new_resource.version(package_provider.package_version)
-          @current_resource
+          if should_download?
```

Comment:
During compile, without the file we don't know the product id (for MSI), so we don't know if the package is installed or not. Aren't we going to always decide that it's not installed and always run the install action during converge?